### PR TITLE
Feature/story upload

### PR DESCRIPTION
**Pull request by AhmedHamdiy**
This pull request introduces the **Stories** feature and a custom hook for managing IndexedDB.

### Stories Feature:
- **AddStory Component:** Enables users to add stories.  
  - Currently supports **image** stories only.
  - Includes the ability to add captions.
  - Features an advanced **image editor** for enhancing and processing images.
- **View and Interaction:**
  - Users can view others' stories.
  - Users can see a list of viewers for their stories.
  - Navigation is simplified with **keyboard controls** (left/right arrows) or on-screen buttons.
  - Stories can be **paused, played, or deleted**.
  - Introduces a visually **avatar outline** for others, indicating the number of unviewed stories.
- **Mocking:** All feature parts are mocked for testing and development.

### IndexedDB Hook:
- A custom hook has been implemented with `setItem` and `getItem` functions for seamless access and storage of values in IndexedDB.

### CODE REVIEW

File: app/src/components/ImageEditor.tsx [hunk 1]
```diff
@@ -30,6 +45,8 @@ function ImageEditor({
   src,
   onImageSave,
   isProfileImage,
+  widthRatio = 50,
+  heightRatio = 50,
 }: imageEditorProps) {
   const handleSave = async (editedImageObject: { imageBase64?: string }) => {
     if (editedImageObject.imageBase64) {
```
Comment (AbdelruhmanSamy) [inline app/src/components/ImageEditor.tsx:52]: Why not using react query in the request?

File: app/src/components/side-bar/chats/ChatsSideBar.tsx [hunk 2]
```diff
@@ -11,27 +12,31 @@ interface ChatsSideBarProps {
...
+  justify-content: space-around;
+  align-items: end;
+  flex-direction: column-reverse;
+  gap: 4rem;
+`;
 function ChatsSideBar({ children }: ChatsSideBarProps) {
   return (
     <StyledChatsSideBar>
       <ChatsSidebarHeader />
+      <StoryListContainer />
```
Comment (AbdelruhmanSamy) [inline app/src/components/side-bar/chats/ChatsSideBar.tsx:34]: Story list is covering search bar, styling needs to be editied

File: app/src/features/authentication/login/services/apiLogin.ts [hunk 3]
```diff
@@ -13,8 +13,9 @@ export async function login(user: User) {
   });
 
   const data = await res.json();
-
+  console.log(data);
```
Comment (AbdelruhmanSamy) [inline app/src/features/authentication/login/services/apiLogin.ts:16]: remove unused printing

File: app/src/features/stories/components/Story.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import { MOBILE_VIEW, STATIC_MEDIA_URL } from "@constants";
```
Comment (AbdelruhmanSamy) [inline app/src/features/stories/components/Story.tsx:1]: Remove unused imports

**Review by AbdelruhmanSamy — DISMISSED** [review]
Absolutely Fantastic, 3aaaash ya hamdyy :fire: :fire:

**Comment by AbdelruhmanSamy** [comment]
image
stories is covering search bar

**Comment by AbdelruhmanSamy** [comment]
image
user info is covering stories

**Comment by AbdelruhmanSamy** [comment]
Story suddenly exits before time is finished upon reviewing it for the second time

File: app/src/components/ImageEditor.tsx (hunk 1 above)
Comment (AhmedHamdiy) [reply app/src/components/ImageEditor.tsx:52]: This component is a generic tool for image editing. This request converts an image to base64 format. I then use React Query for the requests, such as updating a profile picture or sending stories.

File: app/src/features/authentication/login/services/apiLogin.ts (hunk 3 above)
Comment (AhmedHamdiy) [reply app/src/features/authentication/login/services/apiLogin.ts:16]: ana eh elly mkhly apiLogin edited 3ndi asln XD.

ok, I'll remove it.

File: app/src/components/side-bar/chats/ChatsSideBar.tsx (hunk 2 above)
Comment (AhmedHamdiy) [reply app/src/components/side-bar/chats/ChatsSideBar.tsx:34]: Sure

**Review by amir-kedis — DISMISSED** [review]

**Review by amir-kedis — APPROVED** [review]

**Pull request merged**